packages/data/addon/services/navi-dimension.ts

```diff
@@ -22,7 +30,74 @@ export type ServiceOptions = {
   clientId?: string;
 };
 
+type RequestType = 'all' | 'search' | 'find';
+
+/**
+ * a cache specifically for dimension results and queries
+ * a normal cache follows logic "give me key X and I'll give you result Y"
+ * dimension cache, however, is "give me query X and I'll format result Y from what I have"
+ */
+class DimensionCache extends Cache<NaviDimensionResponse> {
+  // set this.getCacheKey() function
+  constructor() {
+    function getDimensionCacheId(dimension: DimensionColumn): string {
+      const cannonicalName = canonicalizeMetric({
+        metric: dimension?.columnMetadata?.name,
+        parameters: dimension.parameters,
+      });
+      return cannonicalName;
+    }
+    super(getDimensionCacheId, MAX_CACHE_SIZE, CACHE_TIMEOUT);
+  }
+
+  /**
+   * pares an 'all' response down into a 'search' response
+   * @param {NaviDimensionResponse} allResponse - the response with all of the dimensions
+   * @param {string} query - the query that you're searching
+   * @returns {NaviDimensionResponse | undefined} the search request dimension request
+   */
+  getSearchFromAll(allResponse: NaviDimensionResponse, query: string): NaviDimensionResponse {
+    return SearchUtils.searchNaviDimensionRecords(A(allResponse.values), query);
+  }
+
+  checkDimensionCache(
+    requestType: RequestType,
+    dimension: DimensionColumn,
+    query?: string
+  ): NaviDimensionResponse | undefined {
+    let results = undefined;
+
+    // check if all dimension results are stored in cache
+    let cacheId = this.getCacheKey(dimension);
+    let cacheResponse = this.getItem(cacheId);
+    if (cacheResponse) {
+      // if all results are in cache, but are looking for specific search
+      if (requestType === 'search' && query) {
+        results = this.getSearchFromAll(cacheResponse, query);
+      } else {
+        results = cacheResponse;
+      }
+    }
+
+    return results;
+  }
+
+  addDimensionToCache(dimension: DimensionColumn, dimensionResults: NaviDimensionResponse) {
+    let cacheId = this.getCacheKey(dimension);
+    this.setItem(cacheId, dimensionResults);
+  }
+}
+
 export default class NaviDimensionService extends Service {
+  /**
+   * @property {DimensionCache} _dimensionCache - local cache for dimensions
+   */
+  private _dimensionCache = new DimensionCache();
+
+  clearCache(): void {
+    this._dimensionCache = new DimensionCache();
```

Comment:
can the `Cache` class itself have a `clear` method or is it more than just creating a new map?